Update token cli docs to new CLI docs template (#28948)
hashicorp/vault: website/content/partials/cli/token/flags/metadata.mdx

> <a id="token-metadata" />

Reviewer: ```suggestion
<a id="token-flag-metadata" />
```

> `-metadata (k=v: "")`**

Reviewer: ```suggestion
`-metadata (string : "")`**
```
The type is a string. The format of the string is "key=value"

> Arbitrary key=value metadata to associate with the token. This metadata will show 

Reviewer: ```suggestion
Custom metadata associated with the token in `"key=value"` format. The metadata shows up
```

in the audit log when the token is used. You can specify this flag multiple times 
to add multiple pieces of metadata.

> **Example**: `-metadata=env=dev`

Reviewer: ```suggestion
**Example**: `-metadata "environment=sandbox"`
```

We've overused the string "dev" in the examples
